src/sentry/eventstream/kafka/backend.py

```diff
@@ -181,38 +183,45 @@ def commit_offsets():
                 )
                 commit(offsets_to_commit)
 
-        try:
-            i = 0
-            while True:
-                message = consumer.poll(0.1)
-                if message is None:
-                    continue
+        shutdown_requested = False
 
-                error = message.error()
-                if error is not None:
-                    raise Exception(error)
+        def handle_shutdown_request(signum: int, frame: Any) -> None:
+            nonlocal shutdown_requested
+            logger.debug("Received signal %r, requesting shutdown...", signum)
```

Comment:
only feedback id have is if this happens irregularly, maybe make this _at least_ info